src/java.base/share/classes/java/lang/reflect/code/bytecode/BytecodeLift.java

```diff
@@ -418,73 +122,332 @@ public static CoreOps.FuncOp liftToBytecodeDialect(byte[] classdata, String meth
+                        case LocalVariable lv -> {
+                            // @@@
+                        }
+                        case LocalVariableType lvt -> {
+                            // @@@
+                        }
+                        case LoadInstruction inst -> {
+                            stack.push(b.op(CoreOps.varLoad(locals.get(inst.slot()))));
+                        }
+                        case StoreInstruction inst -> {
+                            Value operand = stack.pop();
+                            Op.Result local = locals.get(inst.slot());
+                            if (local == null) {
+                                local = b.op(CoreOps.var(Integer.toString(lvm), operand));
+                                locals.put(lvm++, local);
+                            } else {
+                                TypeDesc varType = ((CoreOps.VarOp) local.op()).varType();
+                                if (!operand.type().equals(varType)) {
+                                    local = b.op(CoreOps.var(Integer.toString(lvm), operand));
+                                    locals.put(lvm++, local);
+                                } else {
+                                    b.op(CoreOps.varStore(local, operand));
+                                }
+                            }
+                        }
+                        case IncrementInstruction inst -> {
+                            Op.Result local = locals.get(inst.slot());
+                            b.op(CoreOps.varStore(local, b.op(CoreOps.add(
+                                    b.op(CoreOps.varLoad(local)),
+                                    b.op(CoreOps.constant(TypeDesc.INT, inst.constant()))))));
+                        }
+                        case ConstantInstruction.LoadConstantInstruction inst -> {
+                            stack.push(b.op(switch (inst.constantValue()) {
+                                case ClassDesc v -> CoreOps.constant(TypeDesc.J_L_CLASS, TypeDesc.ofNominalDescriptor(v));
+                                case Double v -> CoreOps.constant(TypeDesc.DOUBLE, v);
+                                case Float v -> CoreOps.constant(TypeDesc.FLOAT, v);
+                                case Integer v -> CoreOps.constant(TypeDesc.INT, v);
+                                case Long v -> CoreOps.constant(TypeDesc.LONG, v);
+                                case String v -> CoreOps.constant(TypeDesc.J_L_STRING, v);
+                                default ->
+                                    // @@@ MethodType, MethodHandle, ConstantDynamic
+                                    throw new IllegalArgumentException("Unsupported constant value: " + inst.constantValue());
+                            }));
+                        }
+                        case ConstantInstruction inst -> {
+                            Op.Result result = b.op(CoreOps.constant(TypeDesc.INT, inst.constantValue()));
```

Comment:
Missing support for float, long, and double, and null constants. Switch on type kind?

Reply:
This case is obsolete, it should be handled by the previous case (unrestricted to LDC).
Will fix it, thanks.